### ISSUE TITLE
YouTube download link don't refresh

### ISSUE THREAD

**Issue by AnAverageBeing** [Sangwan5688/BlackHole#726]
So when u like a song it will save all info like url and download url but for example if I try to redownload that same song after 2 days the link is expired so it downloads a 0 byte file.

I think we should refresh download link before downloading

**Comment by AnAverageBeing** [comment]
I am sorry if that's not how it works I never looked into the code but the problem is I can download songs from yt if they not in my favourite list but if I have some in my list I can't download them it downloads a 0B file

**Comment by Sangwan5688** [comment]
Yup, that's how it behaves. For the time being you can play that song first, app will then refresh the url, and then you can download
